src/main/java/com/suse/saltstack/netapi/event/EventStream.java

```
@@ -183,30 +199,58 @@ public void onOpen(Session session, EndpointConfig config) throws IOException {
+     * @param last indicate the last part of a message
+     * @throws MessageTooBigException in case the message is longer than maxMessageLength
      */
     @OnMessage
-    public void onMessage(String message) {
-        if (message != null && !message.equals("server received message")) {
-            // Salt API adds a "data: " prefix that we need to ignore
-            Event event = JsonParser.EVENTS.parse(message.substring(6));
-            synchronized (listeners) {
-                listeners.stream().forEach(l -> l.notify(event));
+    public void onMessage(String partialMessage, boolean last)
+            throws MessageTooBigException {
+        if (last) {
+            String message;
+            if (partialMessageBuffer.length() == 0) {
+                message = partialMessage;
+            } else {
+                partialMessageBuffer.append(partialMessage);
+                message = partialMessageBuffer.toString();
+
+                // Reset the size to the defaultBufferSize and empty the buffer
+                partialMessageBuffer.setLength(defaultBufferSize);
+                partialMessageBuffer.trimToSize();
+                partialMessageBuffer.setLength(0);
+            }
+            if (maxMessageLength > 0 && message.length() > maxMessageLength) {
+                throw new MessageTooBigException(maxMessageLength);
+            }
+
+            // Notify all registered listeners
+            if (!message.equals("server received message")) {
+                // Salt API adds a "data: " prefix that we need to ignore
+                Event event = JsonParser.EVENTS.parse(message.substring(6));
+                synchronized (listeners) {
+                    listeners.stream().forEach(listener -> listener.notify(event));
+                }
+            }
+        } else {
+            partialMessageBuffer.append(partialMessage);
+            if (maxMessageLength > 0 && partialMessageBuffer.length() > maxMessageLength) {
```

@isbm: Not really. Note that setting `maxMessageLength` to a value `<= 0` is supposed to disable the limit. We never want to throw an exception in this case so there is no need to compare anything. Look at your code: if `maxMessageLength` is set to 0, it would throw an exception for every message with at least one character!

@AdamMajer: I see what you mean, will try to de-duplicate in that way.